main_test.go

```diff
@@ -24,20 +24,24 @@ func TestMain(m *testing.M) {
 	flag.Parse() // NOTE: this is needed to parse args from go test command
 
 	port := func() string { // Get a free port to run the server
-		listener, err := net.Listen("tcp", ":0")
+		listener, err := net.Listen("tcp", "localhost:0")
```

Comment:
```suggestion
		listener, err := net.Listen("tcp", ":0")
```

Using ":0" is generally the most reliable approach as it:

1. Works across all operating systems
2. Doesn't depend on hostname resolution
3. Will work in containerized environments